tiledb/sm/filesystem/ls_scanner.h

```diff
@@ -142,7 +169,9 @@ class LsScanIterator {
    * @return Reference to this iterator after advancing to the next object.
    */
   LsScanIterator& operator++() {
-    scanner_->next(ptr_);
+    if (++ptr_ != pointer()) {
```

Comment:
This looks like a double increment, since `next()` is being called below. I'll let it pass for now since the unit tests pass, but this might be defective.